### PR TITLE
Proposal for patching SonarQube rule 3032

### PR DESCRIPTION
**Pull request by henry-lp**
Signed-off-by: henry-lp <tailp@kth.se>

**Related Issue**
None

**Description of the solution adopted**
This PR fixes the violations for the following SonarQube rules:
https://rules.sonarsource.com/java/RSPEC-3032
If you do no want to receive automated PRs for SonarQube warnings, reply to this PR with 'STOP'

**Screenshots**
None

**Any side note on the changes made**
None

### CODE REVIEW

File: commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java [hunk 1]
```diff
@@ -89,8 +89,8 @@ public int executeUpdate(EntityManager entityManager) {
             StringBuilder sqlStringBuilder = new StringBuilder();
             try {
                 try (
-                        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(qStr);
-                        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
+                    InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(qStr);
```
Comment (LeoNerdoG) [inline commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java:92]: Please update the header to "2017, 2020".

Comment (henry-lp) [reply commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java:92]: Hi, do you perhaps mean at this [line](https://github.com/eclipse/kapua/blob/5246cd29e5ea87aee71374f262d6d07c38071b2d/commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java#L2) ?

Comment (LeoNerdoG) [reply commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java:92]: yes, exactly! 😄

**Comment by LeoNerdoG** [comment]
hey @henry-lp,
when trying to build your branch, I discovered that you have some checkstyle violations:
Screenshot 2020-07-17 at 15 32 48

Output is: 

[ERROR] commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java:[2] (header) RegexpHeader: Line does not match expected header line of '^ \* Copyright \(c\) \d{4}(|, \d{4}) (.*) and others\.?$'.


[ERROR] rest-api/resources/src/test/java/org/eclipse/kapua/app/api/resources/v1/resources/TagsTests.java:[1] (header) RegexpHeader: Line does not match expected header line of '^/\*{79}$'.


[ERROR] rest-api/resources/src/test/java/org/eclipse/kapua/app/api/resources/v1/resources/TagsTests.java:[11,27] (whitespace) MethodParamPad: '(' is preceded with whitespace.


[ERROR] rest-api/resources/src/test/java/org/eclipse/kapua/app/api/resources/v1/resources/TagsTests.java:[14] (whitespace) EmptyLineSeparator: There is more than 1 empty line one after another.

can you please check? Just run `mvn clean install -DskipTests` on your branch.

**Comment by henry-lp** [comment]
@LeoNerdoG ,
Thanks for pointing out :) . For some reasons I only see this checkstyle error at my local branch after pulling the newest changes
> [ERROR] commons/src/main/java/org/eclipse/kapua/commons/jpa/ResourceSqlScriptExecutor.java:[2] (header) RegexpHeader: Line does not match expected header line of '^ * Copyright (c) \d{4}(|, \d{4}) (.*) and others.?$'.

After fixing it I ran `mvn clean install -DskipTests` and it seems to be green at least.

Screenshot from 2020-07-17 15-44-13

**Comment by Coduz** [comment]
Hi @henry-lp ,

Thanks for the contribution.
You may want to add your name to the Contributor in the header of the files? 

Like:
```
/*******************************************************************************
 * Copyright (c) 2017, 2020 Eurotech and/or its affiliates and others
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Eurotech - initial API and implementation
 *     <your name> - <your email>
 *******************************************************************************/
```

I would also suggest to squash your commit into one, since the change is very little but is spread over 3 commits.

Regards, 

_Alberto_

**Comment by henry-lp** [comment]
@Coduz 
> You may want to add your name to the Contributor in the header of the files?

Thanks a lot for the offer. Perhaps somewhere in the future, since I feel like this is only a small change from my part :) . 

Anyways, I have squashed them into one commit as you suggested and also, thanks for accepting this patch. 

Best,
Henry

**Review by Coduz — APPROVED** [review]

**Comment by Coduz** [comment]
Hi @henry-lp ,

this fix has been back-ported to `release-1.2.x` branch so it will be released in Kapua `1.2.5`.

Thanks, 

_Alberto_

**Pull request merged**